src/PDO/Peachpie.Library.PDO/PDOStatement.cs

```diff
@@ -1082,6 +1082,12 @@ public PhpValue fetchColumn(int column_number = 0)
                 return PhpValue.False;
             }
 
+            // read next row
+            if (!this.m_dr.Read())
```

Comment:
are we sure we want to read **next** row for every `fetchColumn` ?

Reply:
According to the PHP documentation it should fetch the next row:
https://www.php.net/manual/en/pdostatement.fetchcolumn.php

`PDOStatement::fetchColumn — Returns a single column from the next row of a result set`
`Returns a single column from the next row of a result set or FALSE if there are no more rows.`

So I interpreted that as it will always fetch a new row. I'll try to do a test in PHP to see how it behaves to confirm

Reply:
thank you, I was confused a little

Reply:
Just to be safe I ran the following test in PHP:
```php
$statement = $pdo->prepare("SELECT * FROM tasks");
$statement->execute();

$tmp1 = $statement->fetchColumn(1);
$tmp2 = $statement->fetchColumn(1);
$tmp3 = $statement->fetchColumn(1);

echo $tmp1 . "\r\n";
echo $tmp2 . "\r\n";
echo $tmp3 . "\r\n";
```

Where the tasks table is filled like this:
```
  id  |  name  |
  1   |  row1  |
  2   |  row2  |
  3   |  row3  |
```

The output in PHP is:
```
row1
row2
row3
```

Reply:
thanks, I've checked the sample in PHP's manual as well and you're right